pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -556,13 +555,20 @@ func (jm *ControllerV2) syncCronJob(
 	case errors.HasStatusCause(err, corev1.NamespaceTerminatingCause):
 	case errors.IsAlreadyExists(err):
 		klog.InfoS("Job already exists", "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()), "job", klog.KRef(jobReq.GetNamespace(), jobReq.GetName()))
+		var getError error
+		jobResp, getError = jm.jobControl.GetJob(jobReq.Namespace, jobReq.Name)
+		if getError != nil {
+			return cj, nil, getError
+		}
 	case err != nil:
 		// default error handling
 		jm.recorder.Eventf(cj, corev1.EventTypeWarning, "FailedCreate", "Error creating job: %v", err)
 		return cj, nil, err
+	default:
+		// error is nil, job was just created
+		klog.V(4).InfoS("Created Job", "job", klog.KRef(jobResp.GetNamespace(), jobResp.GetName()), "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()))
+		jm.recorder.Eventf(cj, corev1.EventTypeNormal, "SuccessfulCreate", "Created job %v", jobResp.Name)
 	}
-	klog.V(4).InfoS("Created Job", "job", klog.KRef(jobResp.GetNamespace(), jobResp.GetName()), "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()))
-	jm.recorder.Eventf(cj, corev1.EventTypeNormal, "SuccessfulCreate", "Created job %v", jobResp.Name)
```

Comment:
Based on my previous comment, that should not be needed.

Reply:
+1 dropped the change